app/src/main/java/woowacourse/shopping/view/cart/vm/CartViewModelFactory.kt

```
@@ -0,0 +1,17 @@
+import androidx.lifecycle.ViewModelProvider
+import woowacourse.shopping.data.CartStorage
+import woowacourse.shopping.data.CartStorageImpl
+
+class CartViewModelFactory : ViewModelProvider.Factory {
+    override fun <T : ViewModel> create(modelClass: Class<T>): T {
+        val storage: CartStorage = CartStorageImpl
+
+        if (modelClass.isAssignableFrom(CartViewModel::class.java)) {
+            return CartViewModel(storage) as T
+        }
+        throw IllegalArgumentException()
```

detekt (1.23.7)

[warning] A call to the default constructor of an exception was detected. Instead one of the constructor overloads should be called. This allows to provide more meaningful exceptions.

(detekt.exceptions.ThrowingExceptionsWithoutMessageOrCause)